homestar-runtime/src/event_handler/notification/network.rs

```diff
@@ -0,0 +1,2671 @@
+    SentWorkflowInfo(SentWorkflowInfo),
+    /// Received workflow info notification.
+    #[schemars(rename = "received_workflow_info")]
+    ReceivedWorkflowInfo(ReceivedWorkflowInfo),
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) enum WorkflowInfoSource {
+    Dht,
+    RequestResponse,
+}
+
+impl fmt::Display for NetworkNotification {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match *self {
+            NetworkNotification::NewListenAddr(_) => write!(f, "new_listen_addr"),
+            NetworkNotification::ConnnectionEstablished(_) => write!(f, "connection_established"),
+            NetworkNotification::ConnnectionClosed(_) => write!(f, "connection_closed"),
+            NetworkNotification::OutgoingConnectionError(_) => {
+                write!(f, "outgoing_connection_error")
+            }
+            NetworkNotification::IncomingConnectionError(_) => {
+                write!(f, "incoming_connection_error")
+            }
+            NetworkNotification::DiscoveredMdns(_) => write!(f, "discovered_mdns"),
+            NetworkNotification::DiscoveredRendezvous(_) => write!(f, "discovered_rendezvous"),
+            NetworkNotification::RegisteredRendezvous(_) => write!(f, "registered_rendezvous"),
+            NetworkNotification::DiscoverServedRendezvous(_) => {
+                write!(f, "discover_served_rendezvous")
+            }
+            NetworkNotification::PeerRegisteredRendezvous(_) => {
+                write!(f, "peer_registered_rendezvous")
+            }
+            NetworkNotification::PublishedReceiptPubsub(_) => write!(f, "published_receipt_pubsub"),
+            NetworkNotification::ReceivedReceiptPubsub(_) => write!(f, "received_receipt_pubsub"),
+            NetworkNotification::PutReceiptDht(_) => write!(f, "put_receipt_dht"),
+            NetworkNotification::GotReceiptDht(_) => write!(f, "got_receipt_dht"),
+            NetworkNotification::PutWorkflowInfoDht(_) => write!(f, "put_workflow_info_dht"),
+            NetworkNotification::GotWorkflowInfoDht(_) => write!(f, "got_workflow_info_dht"),
+            NetworkNotification::ReceiptQuorumSuccessDht(_) => {
+                write!(f, "receipt_quorum_success_dht")
+            }
+            NetworkNotification::ReceiptQuorumFailureDht(_) => {
+                write!(f, "receipt_quorum_failure_dht")
+            }
+            NetworkNotification::WorkflowInfoQuorumSuccessDht(_) => {
+                write!(f, "workflow_info_quorum_success_dht")
+            }
+            NetworkNotification::WorkflowInfoQuorumFailureDht(_) => {
+                write!(f, "workflow_info_quorum_failure_dht")
+            }
+            NetworkNotification::SentWorkflowInfo(_) => {
+                write!(f, "sent_workflow_info")
+            }
+            NetworkNotification::ReceivedWorkflowInfo(_) => {
+                write!(f, "received_workflow_info")
+            }
+        }
+    }
+}
+
+impl DagJson for NetworkNotification {}
```

Comment:
Yeah, @bgins I vote we at least break this up in this PR across some subset of files. It's really too much in one place.

Reply:
Yeah, good point. Re-factored this to break the network notifications into modules. 👍